### DISCUSSION TITLE
Is it possible to link purchased plugin licences to a Craft Cloud Craft CMS licence?

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [craftcms/cloud#48] [Q&A]
Is there a way to claim the Craft Cloud Craft CMS licence to link plugins to? The reason I ask is because we have plugins from a previous licence which we now in use on Craft Cloud, however they keep getting linked back to the previous licence. I have unlinked the Craft CMS licence from them in console several times but they keep being added back to the previous CMS licence. We no longer have any public environment running for the old site (there was a 24 hour period where this was the case), but a local development environment does still exist and has been spun up through DDEV recently, I'm assuming this is potentially triggering the behaviour. I have blanked the plugin licence environment variables in use on the old site, to stop this potentially, but I did wonder if you can link plugins to the Craft Cloud licence generally? I couldn't see anywhere you could obtain it to do it though.

**Comment by jasonmccallister** [comment]
This is something we recently discussed internally and have plans to correlate the plugins to the Craft Cloud project, but not sure on the timeline at the moment.

**Comment by angrybrad** [comment]
To be clear, a plugin license is tied to a Craft license when it’s entered into Settings->Plugins in the control panel, and that eventually hits a “public" domain.

That applies to both self-hosted Craft installs and Cloud.

What is missing is a way to _see_ the correlations between Craft and plugin licenses on Craft Console for Cloud-based Craft licenses.

> The reason I ask is because we have plugins from a previous licence which we now in use on Craft Cloud, however they keep getting linked back to the previous licence. I have unlinked the Craft CMS licence from them in console several times but they keep being added back to the previous CMS licence.

Something else must be going on there. Can you send an email to support@craftcms.com with the relevant license keys, domains, etc., and we can investigate?

**Comment by jamesmacwhite** [comment]
@angrybrad @jasonmccallister Thanks both!

@angrybrad I think the issue might have been the local environment where the environment variables were set to the licences on the old site. I have since blanked them on the development environment and thus far looking at Craft Console, the licences remain unlinked and not attached. I assume this was the likely reason, given the local development environment had the Craft CMS licence key it kept being linked back to.
